test-framework/core/src/main/java/org/keycloak/test/framework/admin/KeycloakAdminClientSupplier.java

```diff
@@ -25,14 +31,46 @@ public Class<Keycloak> getValueType() {
 
     @Override
     public Keycloak getValue(InstanceContext<Keycloak, InjectAdminClient> instanceContext) {
+        InjectAdminClient annotation = instanceContext.getAnnotation();
+
+        InjectAdminClient.Mode mode = annotation.mode();
+
         KeycloakServer server = instanceContext.getDependency(KeycloakServer.class);
-        return KeycloakBuilder.builder()
+        KeycloakBuilder clientBuilder = KeycloakBuilder.builder()
                 .serverUrl(server.getBaseUrl())
-                .realm("master")
-                .grantType(OAuth2Constants.CLIENT_CREDENTIALS)
-                .clientId(Config.getAdminClientId())
-                .clientSecret(Config.getAdminClientSecret())
-                .build();
+                .grantType(OAuth2Constants.CLIENT_CREDENTIALS);
+
+        if (mode.equals(InjectAdminClient.Mode.BOOTSTRAP)) {
+            clientBuilder.realm("master").clientId(Config.getAdminClientId()).clientSecret(Config.getAdminClientSecret());
+        } else if (mode.equals(InjectAdminClient.Mode.MANAGED_REALM)) {
+            ManagedRealm managedRealm = instanceContext.getDependency(ManagedRealm.class);
+            clientBuilder.realm(managedRealm.getName());
+
+            String clientId = !annotation.client().isEmpty() ? annotation.client() : null;
+            String userId = !annotation.user().isEmpty() ? annotation.user() : null;
+
+            if (clientId == null) {
```

Comment:
@stianst to me it seems redundant, since below you are checking both: client and user?